onnxruntime/core/providers/cpu/tensor/scatter.cc

```diff
@@ -106,7 +106,7 @@ Status CopyScatterData(const Tensor* data_input, const Tensor* indices_input, co
     // We start at num_dims - 2 because we already pre-populated
     // the last element above
     for (int64_t i = int64_t(num_dims - 2); i >= 0; --i) {
-      dim_block_size[i] = input_data_shape[i] * dim_block_size[i + 1];
+      dim_block_size[i] = input_data_shape[i + 1] * dim_block_size[i + 1];
```

Comment:
> dim_block_size[i] = input_data_shape[i + 1] * dim_block_size[i + 1]; [](start = 5, length = 69)

You change seems to be correct. However, the current version of the unit test that you are referring to does not repro the assert. Thus as it is there is not a verification that the assert is indeed fixed.
Your WithAxis test is different. Please, add it with a different name to demonstrate the assert and then fix it.

Reply:
I see that you added it already under different name.

---
In reply to: [274102583](https://github.com/Microsoft/onnxruntime/pull/628#discussion_r274102583) [](ancestors = 274102583)